c7n/resources/waf.py

```diff
@@ -87,3 +89,52 @@ class resource_type(TypeInfo):
         'describe': DescribeWafV2,
         'config': ConfigSource
     }
+
+
+@WAFV2.filter_registry.register('logging')
+class WAFV2LoggingFilter(ValueFilter):
+    """
+    Filter by wafv2 logging configuration
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: wafv2-logging-enabled
+            resource: aws.wafv2
+            filters:
+              - not:
+                  - type: logging
+                    key: ResourceArn
+                    value: present
+
+          - name: check-redacted-fields
+            resource: aws.wafv2
+            filters:
+              - type: logging
+                key: RedactedFields[].SingleHeader.Name
+                value: user-agent
+                op: in
+                value_type: swap
+    """
+
+    schema = type_schema('logging', rinherit=ValueFilter.schema)
+    permissions = ('wafv2:GetLoggingConfiguration', )
+    annotation_key = 'c7n:WafV2LoggingConfiguration'
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client(
+            'wafv2', region_name=self.manager.region)
+        for r in resources:
+            if not r.get(self.annotation_key):
+                try:
+                    logging_conf = client.get_logging_configuration(
```

Comment:
It looks like [`list_logging_configurations()`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/wafv2.html#WAFV2.Client.list_logging_configurations) returns the same fields as [`get_logging_configuration()`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/wafv2.html#WAFV2.Client.get_logging_configuration). We may be able to use that (w/`REGIONAL` scope to match the wafv2 describe source) to get the same data in fewer calls.

Reply:
thanks, updated